docs/ui/components/Link/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { css } from '@emotion/react';
+import { theme } from '@expo/styleguide';
+import NextLink from 'next/link';
+import React, { forwardRef } from 'react';
+
+export type LinkProps = React.PropsWithChildren<{
+  target?: string;
+  tabIndex?: number;
+  href?: string;
+  title?: string;
+  rel?: string;
+  as?: string;
+  isStyled?: boolean;
+  onClick?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+  onMouseMove?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+  onMouseEnter?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+  onMouseLeave?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+  onTouchStart?: (event: React.TouchEvent<HTMLAnchorElement>) => void;
+  testID?: string;
+  style?: React.CSSProperties;
+  className?: string;
+  openInNewTab?: boolean;
+  ariaLabel?: string;
+}>;
+
+export const Link = forwardRef<HTMLAnchorElement, LinkProps>(function Link(props, ref) {
+  const { href, openInNewTab = false } = props;
+  const relProp = props.target === '_blank' && !props.rel ? 'noopener' : props.rel;
+
+  function onKeyDown(event: React.KeyboardEvent<HTMLAnchorElement>) {
+    if (event.key === 'Enter' && props.onClick) {
+      props.onClick(event as any);
+    }
+  }
+
+  return (
+    <NextLink href={href ?? ''} passHref={Boolean(href)}>
+      <a
+        ref={ref}
+        aria-label={props.ariaLabel}
+        css={props.isStyled ? linkStyle : undefined}
+        className={props.className}
+        title={props.title}
+        style={props.style}
+        onClick={props.onClick}
+        tabIndex={props.tabIndex}
+        data-testid={props.testID}
+        target={openInNewTab ? '_blank' : props.target}
+        rel={openInNewTab ? 'noopener noreferrer' : relProp}
+        onMouseMove={props.onMouseMove}
+        onMouseEnter={props.onMouseEnter}
+        onMouseLeave={props.onMouseLeave}
+        onTouchStart={props.onTouchStart}
+        onKeyDown={onKeyDown}>
```

Comment:
```suggestion
      <a
        ref={ref}
        aria-label={props.ariaLabel}
        css={props.isStyled ? linkStyle : undefined}
        className={props.className}
        title={props.title}
        style={props.style}
        onClick={props.onClick}
        tabIndex={props.tabIndex}
        data-testid={props.testID}
        target={openInNewTab ? '_blank' : props.target}
        rel={openInNewTab ? 'noopener noreferrer' : relProp}
      >
```

Probably don't need to list out these props. This component in the website could also use a refactor.

Reply:
Added a task for website, ENG-4033